File: texture-generator/pre/colorfilter/InverseColorFilter.ts
class InverseColorFilter extends MultiColorFilter {

    constructor(colorLow: string, colorMid: string, colorHigh: string){
        super(colorLow, colorMid, colorHigh);
    }

    colorPixel(pixel: number, index: number): number {
        const a = ColorFilter.getA(pixel);
        if(a === 0){
            return pixel;
        }
        const color = this.brightnessData[index] < this.minBrightness ? this.colorLow : this.brightnessData[index] > this.maxBrightness ? this.colorHigh : this.colorMid;
        const r = ~ColorFilter.mult(ColorFilter.getR(color), this.brightnessData[index]) & 0xff;
        const g = ~ColorFilter.mult(ColorFilter.getG(color), this.brightnessData[index]) & 0xff;
        const b = ~ColorFilter.mult(ColorFilter.getB(color), this.brightnessData[index]) & 0xff;
        return Color.argb(a, r, g, b);
    }

}

File: texture-generator/filters/inverse_color_filter.py
from . import MultiColorFilter, RGBA
import utils


class InverseColorFilter(MultiColorFilter):

    def color_pixel(self, rgba: RGBA, index: int) -> RGBA:

        if rgba[3] == 0:
            return rgba
        
        color: RGBA
        if self.brightness_data[index] < self.brightness_min:
            color = self.color_low
        elif self.brightness_data[index] > self.brightness_max:
            color = self.color_high
        else:
            color = self.color_mid

        r = ~utils.mult_color(color[0], self.brightness_data[index]) & 0xff
        g = ~utils.mult_color(color[1], self.brightness_data[index]) & 0xff
        b = ~utils.mult_color(color[2], self.brightness_data[index]) & 0xff

        return (r, g, b, rgba[3])
